pymatgen/io/vasp/sets.py

```diff
@@ -1215,6 +1215,73 @@ def from_prev_calc(cls, prev_calc_dir, **kwargs):
         return input_set.override_from_prev_calc(prev_calc_dir=prev_calc_dir)
 
 
+class MatPESStaticSet(MPStaticSet):
+    """Creates input files for a MatPES static calculation."""
+
+    CONFIG = _load_yaml_config("MatPESStaticSet")
+
+    def __init__(
+        self,
+        structure,
+        functional=None,
+        **kwargs,
+    ):
+        """
+        Args:
+            structure (Structure): Structure from previous run.
+            functional (str): We can set the VASP functional to R2SCAN or R2SCAN+U or PBE+U, or it will
+                generate input sets PBE functional.
+            **kwargs: kwargs supported by MPRelaxSet.
+        """
+        super().__init__(structure, MatPESStaticSet.CONFIG, **kwargs)
+        kwargs.setdefault("user_potcar_functional", "PBE_54")
+        if isinstance(functional, str):
+            if functional == "R2SCAN":
+                self.user_incar_settings = {"METAGGA": "R2SCAN"}
+            if functional == "R2SCAN+U":
+                self.user_incar_settings = {"METAGGA": "R2SCAN", "LDAU": True}
+            if functional == "PBE+U":
+                self.user_incar_settings = {"LDAU": True}
+
+        self.kwargs = kwargs
+        self.functional = functional
+
+    @property
+    def incar(self):
+        """Incar"""
+        parent_incar = super().incar
+        incar = Incar(self.prev_incar) if self.prev_incar is not None else Incar(parent_incar)
+
+        incar.update(
```

Comment:
I am confused why this section is needed? What is the reason to specify the settings in the YAML file, but still have to hard code this in the input set? If anything, it should be incar.update(parent_incar) to override all these settings from prev_incar if there is any. Ultimately, all these will need to have tests in order for us to know if the code is correctly working.